doc/advanced_data_structures.rst

```diff
@@ -0,0 +1,117 @@
+******************************
+Advanced Data Structure Topics
+******************************
+
+This page of documentation is reserved for advanced topics in SmartRedis
+data structures that may not be needed for all users.
+
+.. _advanced_data_structures_dataset_aggregation:
+
+Dataset Aggregation
+===================
+
+In addition to the ability to work with individual datasets, SmartRedis offers
+users the manage lists of datasets and work with them collectively. This is
+particularly useful for producer/consumer relationships.
+
+The DataSet Aggregation API stores multiple ``DataSet`` objects
```

Comment:
Maybe change to ``The DataSet Aggregation API manages references to multiple ``DataSet`` objects``

Reply:
More precise, I like it. Done